nkms/crypto/utils.py

```diff
@@ -45,9 +85,35 @@ def __call__(self, splittable, return_remainder=False, msgpack_remainder=False):
     def __len__(self):
         return sum(self.get_message_meta(m)[1] for m in self.message_types)
 
+
     @staticmethod
     def get_message_meta(message_type):
-        return message_type if isinstance(message_type, tuple) else (message_type, message_type._EXPECTED_LENGTH)
+        try:
```

Comment:
Probably ok for now, but looks like quite a bit of try/except magic is going on. In future, I'd consider looking at this place again to reduce possibilities of passing the wrong message_type which produces a non-error output

Reply:
Agreed, we should probably be more clear with what's going on here. It's not inherently clear to me what each `try`  block is trying to do first.

Reply:
Perhaps another in-line comment or two on the subscriptions, but I actually think this is a fairly elegant and idiomatic way of determining what category of byte-string you have, and gives the user info they need to use ByteStingSplitter correctly.

Reply:
I think I'm with @KPrasch on this one - it's not like it's weird nesting or anything.  It's three try blocks: one for each expected element in the tuple.  

Ask yourself: will you be more comfortable if this is three `ifs`?  If so, I assert that this is more elegant than that.